### DISCUSSION TITLE
ESPHome firmware on Sonoff ZBBridge ESP8266 as a remote Zigbee adapter

### DISCUSSION THREAD

**Discussion by Hedda** [zigpy/zigpy#719] [General]
FYI @thegroove created an "ESPHome for Sonoff Zigbee Bridge" project with prepackaged [ESPHome](https://esphome.io) custom component and configuration for the Sonoff Zigbee Bridge with serial-over-tcp/ip proxy server functionality:

https://github.com/thegroove/esphome-zbbridge

Backstory:

https://github.com/esphome/feature-requests/issues/688

This makes the Sonoff ZBBridge behave more similar to that of Tube's Zigbee Gateways coordinators based on EFR32:

https://github.com/tube0013/tube_gateways/tree/main/tube_zb_gw_efr32

https://github.com/tube0013/tube_gateways

https://www.tubeszb.com/shop/coordinators/2

**Comment by Hedda** [comment]
I would love to see this with a Zigbee Router configuration as being a WiFi bridge it make a better router than a Zigbee coordinator.

If I understand correctly it should also be possible to configure this as a Zigbee router and join an existing network?

If so do you know the serial commands for configuring it as a router and putting it into pairing mode so it can be discovered?

**Reply by MattWestb** [reply]
[UG100: EZSP Reference Guide](https://www.silabs.com/documents/public/user-guides/ug100-ezsp-reference-guide.pdf) is all commands that you can using if you have loading  one EZSP firmware on your NCP.

If you have bellows running you can joining it with easy commands as one sleeping end device but it dont have command for joining as router but i think its easy to implanting of our code master.  
If was having one command for joining as router its possible loading ESPHome and then sending commands with bellows or doing macros in ESPHome that can being fired from  the web UI for doing leave and joining the network with one standard EZSP NCP firmware and no special hard or firmware.

One ESP-02S and one IKEA ICC-A-1 and one buck converter and small USB power supply from one old mobile phone and you is up and running.

**Reply by Hedda** [reply]
FYI, @xsp1989 has now uploaded/shared a signed Zigbee Router firmware for ITead's Sonoff ZBBridge to his google drive here:

> https://drive.google.com/file/d/1H-M5CTh_XGGVl2te4vHLQ5SfGPHL6yCS/view?usp=sharing
> 
> This is the signed routing firmware used by ZBB, the usage method is the same as the unsigned firmware.
> 
> https://github.com/xsp1989/zigbeeFirmware/blob/master/firmware/Zigbee3.0_Dongle/RouterForDongle/README.md

PS: This idea is also being discussed specifically for ITead's Sonoff ZBBridge hacked with Tasmota here -> https://github.com/arendst/Tasmota/discussions/11536

**Reply by Hedda** [reply]
FYI, the same Zigbee Router firmware has now also been tested with ESPHome on a ZB-GW03 eWeLink Ethernet Zigbee Gateway which is based on ESP32 instead but have the same “SM-011 V1.0” EFR32MG21 Zigbee module by CoolKit that ITead uses in their Sonoff ZBBridge Zigbee Bridge:

https://github.com/syssi/esphome-zb-gw03/blob/main/docs/router.md

https://www.digiblur.com/2022/02/convert-sonoff-zigbee-bridge-to-router-repeater.html
